ggml/include/ggml-opt.h

```diff
@@ -74,16 +74,30 @@ extern "C" {
         GGML_OPT_BUILD_TYPE_OPT     = 30,
     };
 
+    enum ggml_opt_optimizer_type {
+        GGML_OPT_OPTIMIZER_ADAMW,
+        GGML_OPT_OPTIMIZER_SGD,
+
+        GGML_OPT_OPTIMIZER_COUNT
+    };
+
+    // "adamw" or "sgd" (case insensitive)
+    GGML_API const char *                 ggml_opt_optimizer_name(enum ggml_opt_optimizer_type);
+    GGML_API enum ggml_opt_optimizer_type ggml_opt_get_optimizer(const char *);
+
     // parameters that control which optimizer is used and how said optimizer tries to find the minimal loss
     struct ggml_opt_optimizer_params {
-        // AdamW optimizer parameters
+        // SGD and AdamW optimizer parameters
         struct {
-            float alpha; // learning rate
-            float beta1;
-            float beta2;
-            float eps;   // epsilon for numerical stability
-            float wd;    // weight decay for AdamW, use 0.0f to disable
+            float alpha;  // learning rate
+            float beta1;  // adamw
+            float beta2;  // adamw
+            float eps;    // epsilon for numerical stability
+            float wd;     // weight decay for SGD or AdamW, use 0.0f to disable
         } adamw;
+
+        // only GGML_OPT_OPTIMIZER_ADAMW allocates m, v per parameter
+        enum ggml_opt_optimizer_type optimizer;
```

Comment:
Remove this from `ggml_opt_optimizer_params` and add it to `ggml_opt_params` instead. The optimizer parameters are fetched once per epoch and having this property here would suggest that it's possible to change the optimizer type during training. But that is going to make the code much more complicated and IIRC we agreed that we would not add support for it.